src/controlplane/scraper.go

```diff
@@ -92,15 +94,35 @@ func NewScraper(config *config.Config, providers Providers, options ...ScraperOp
 		}
 	}
 
+	var err error
 	// TODO If this could change without a restart of the pod we should run it each time we scrape data,
 	// possibly with a reasonable cache Es: NewCachedDiscoveryClientForConfig
 	s.k8sVersion, err = providers.K8s.Discovery().ServerVersion()
 	if err != nil {
 		return nil, fmt.Errorf("fetching K8s version: %w", err)
 	}
 
-	// Building pod lister and closers for pod autodisover.
-	s.buildLister()
+	secretListerer, informerCloser := discovery.NewSecretNamespaceLister(discovery.SecretListerConfig{
+		Client:     s.K8s,
+		Namespaces: secretNamespaces(s.components),
+	})
+
+	s.informerClosers = append(s.informerClosers, informerCloser)
+
+	s.authenticator, err = authenticator.New(
+		authenticator.Config{
+			SecretListerer:  secretListerer,
+			InClusterConfig: s.inClusterConfig,
+		},
+		authenticator.WithLogger(s.logger),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("creating authenticator: %w", err)
+	}
+
+	// TODO refactor pod discovery to add multiple namespaces and implement similar
```

Comment:
would make sense to address it now?

Reply:
yes but i wanted to validate that the `SecretListerer` approach is ok

Reply:
just pushed this change